### PR TITLE
fix: policies registry returns same data types as other registries

### LINKED ISSUE #20893: CLI command `strapi policies:list` is broken
In a Strapi v5 application, the `strapi policies:list` throws an error:

```txt
TypeError: list.forEach is not a function
    at action (/mystrapi/packages/core/strapi/dist/cli/commands/policies/list.js:18:8)
```

You can test in the getstarted directory with this Yarn:

```sh
yarn strapi policies:list
```

### PR DESCRIPTION
**Pull request by innerdvations**
### What does it do?

- fixes policies registry to return the same data types as controllers, services, etc

### Why is it needed?

fixes the `strapi policies:list` command and makes the policies `keys()` return type consistent with other registries

### How to test it?

Policies should all work fine
especially check that `strapi policies:list` should return policies

### Related issue(s)/PR(s)

DX-1505 

Fixes #20893

### CODE REVIEW

File: packages/core/core/src/registries/policies.ts [hunk 1]
```diff
@@ -108,7 +109,22 @@ const policiesRegistry = () => {
      * Returns a map with all the policies in a namespace
      */
     getAll(namespace: string) {
-      return pickBy((_, uid) => hasNamespace(uid, namespace))(Object.fromEntries(policies));
+      const filteredPolicies = pickBy((_, uid) => hasNamespace(uid, namespace))(
+        Object.fromEntries(policies)
+      );
+
+      // Return an enumerable object of the same format as controllers, services, etc
```
Comment (alexandrebodin) [inline packages/core/core/src/registries/policies.ts:116]: I don't see why this is necessary ? the picKBy should already return a enumerable object 🤔

Comment (innerdvations) [reply packages/core/core/src/registries/policies.ts]: Oops, you're right, that one can be left alone I think.

**Comment by alexandrebodin** [comment]
I did not QA anything but the code looks fine. I'm not sure we actually use policies.keys() anywhere ?

**Comment by innerdvations** [comment]
We only use it one place in our own code

**Review by Convly — APPROVED** [review]
LGTM

**Pull request merged**